### PR TITLE
test: add test for getScreenshot and pushNotification commands

### PR DESCRIPTION
**Pull request by mwakizaka**
should be merged after https://github.com/appium/node-simctl/pull/174
I didn't add tests for `addCertificate` command as we roughly cover it in https://github.com/appium/appium-xcuitest-driver/pull/1507

### CODE REVIEW

File: test/e2e/simctl-e2e-specs.js [hunk 1]
```diff
@@ -270,5 +272,30 @@ describe('simctl', function () {
...
+        const payload = {
+          'Simulator Target Bundle': 'com.apple.Preferences',
+          'aps': {
+            'alert': 'This is a simulated notification!',
+            'badge': 3,
+            'sound': 'default'
+          }
+        };
+
+        expect(async () => (await simctl.pushNotification(payload))).to.not.throw();
```
Comment (mykola-mokhnach) [inline test/e2e/simctl-e2e-specs.js:297]: for promises you could also use `should.be.fulfilled`

**Review by mykola-mokhnach — APPROVED** [review]

File: test/e2e/simctl-e2e-specs.js [hunk 2]
```diff
@@ -270,5 +272,30 @@ describe('simctl', function () {
         fullList.runtimes.filter((el) => el.identifier.includes('iOS')).length.should.be.above(0);
       });
     });
+
+    describe('getScreenshot', function() {
+      it('should get a base64 string', async function () {
+        const image = await simctl.getScreenshot();
+
+        // Use a rough regex to check if a returned value is base64 string
+        const regex = new RegExp(/[A-Za-z0-9+/=]/);
```
Comment (mykola-mokhnach) [inline test/e2e/simctl-e2e-specs.js:281]: you don't need to use an explicit RegExp constructor: const regex = `/^[A-Za-z0-9+/=]+$/`

Comment (mykola-mokhnach) [reply test/e2e/simctl-e2e-specs.js:281]: I think something like `Buffer.from(str, 'base64').toString('base64') === str` would better verify base64-encoded entity

**Comment by mykola-mokhnach** [comment]
https://github.com/appium/node-simctl/pull/174 has been merged and published

File: test/e2e/simctl-e2e-specs.js (hunk 1 above)
Comment (mwakizaka) [reply test/e2e/simctl-e2e-specs.js]: thanks, addressed as https://github.com/appium/node-simctl/pull/175/commits/167e91e6a96f8a569dee78938f877b2955c76272

File: test/e2e/simctl-e2e-specs.js (hunk 2 above)
Comment (mwakizaka) [reply test/e2e/simctl-e2e-specs.js]: thank you, addressed as https://github.com/appium/node-simctl/pull/175/commits/b308bce0f39adce61e7eb105fbb086f049072755

File: test/e2e/simctl-e2e-specs.js [hunk 3]
```diff
@@ -267,5 +269,28 @@ describe('simctl', function () {
         fullList.runtimes.filter((el) => el.identifier.includes('iOS')).length.should.be.above(0);
       });
     });
+
+    describe('getScreenshot', function() {
+      it('should get a base64 string', async function () {
+        const image = await simctl.getScreenshot();
+
+        expect(Buffer.from(image, 'base64').toString('base64') === image).to.be.true;
```
Comment (mykola-mokhnach) [inline test/e2e/simctl-e2e-specs.js:277]: you could use .eql or .equal instead of expect there

**Pull request merged**